### PR TITLE
Split release notes file in two parts

### LINKED ISSUE #101714: [Meta] RELEASES.md isn't rendered on GitHub.com
Hi, Rust contributors!

Take a look at Rust releases file: https://github.com/rust-lang/rust/blob/master/RELEASES.md. For me, GitHub just states that rendering it took too long, so it won't show it. In my opinion rendering Markdown shouldn't be so hard to render a rarely updated half-megabyte file, but it's to GitHub developers who are managing the platform and their workload might dictate the limits. However, even if the current file could be rendered, it will still continue to grow with each release.

I see several ways to tackle this problem:
* Keep things as they are right now. Even if Releases isn't viewable on the GitHub website, you can still read it sources or view in editor/IDE that can show rendered MarkDown, and I guess releases blog posts are very similar to a separate chapter of that file.
* Contact GitHub team, maybe they could provide a solution or advice from their side.
* Split RELEASES.md. file. Several other projects use several Releases / ChangeLog files, so that every one of those isn't that big. However, it often comes from the fact that they have a deeper versioning scheme, where every file represents one major or minor series, i.e. `1.*` or `1.2.*`, and are created rarely, while fixes releases are much more often. Rust versions since 1.0 aren't like that, we for the most time just have incrementing minor version with rare bug-fixing releases. I have a quick thought that we could just aggregate those by 10 or by release a year, but it requires further thoughts on UX of the whole thing.
   * What we could do is to add just one "older releases" and only keep several recent ones (like, for a year) in a `RELEASES.md`. 

Anyway, those are just my thoughts, and maybe tomorrow the file will still be viewable as it has been before.

### PR DESCRIPTION
**Pull request by djc**
For a while now, it's been a pain that `RELEASES.md` no longer renders as HTML when browsing GitHub (because it's grown too large) -- [discussion on Zulip](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/Release.20blob.20too.20large). Split the file in two parts, using 1.56 (as the first release containing 2021 edition support) as the boundary -- the old file is now called `RELEASES-old.md`. `RELEASES-old` points to `RELEASES.md` at the top, while `RELEASES.md` now points to `RELEASES-old.md` at the bottom.

Alternatively, we could explicitly call `-old` something more like `-pre-1.56`, which has the advantage of being more clear up front but the disadvantage of having to move it when we next want to move the boundary.

As another alternative, we could split it up over more files (one per edition?) so that each file will be small enough to render even in the longer term. However, that would make it (slightly) harder to search all of them together.

There are different colors we could paint this particular bikeshed but this seems like a nice incremental improvement.

Fixes #101714.

### CODE REVIEW

**Comment by tspiteri** [comment]
> Alternatively, we could explicitly call `-old` something more like `-pre-1.56`, which has the advantage of being more clear up front but the disadvantage of having to move it when we next want to move the boundary.

Why "move" the boundary? Maybe just adding a new file every few years is preferable, so it would be creating a new boundary, not moving the existing one.

**Review by weihanglo — COMMENTED** [review]
Thanks for the MVP! However, this breaks old links everywhere on the internet. We should use our best effort to keep those links alive.

**Comment by djc** [comment]
I would argue those links are currently broken, and this PR doesn't fix (most of) them.

**Comment by weihanglo** [comment]
Off the top of my head, probably have this in `RELEASE.md` for compatibility.

```markdown
Version 1.69.0 (2023-04-20)
==========================

<a id="1.69.0-Language"></a>
<a id="1.69.0-Compiler"></a>
<a id="1.69.0-Libraries"></a>
<a id="1.69.0-Stabilized-APIs"></a>
<a id="1.69.0-Cargo"></a>
<a id="1.69.0-Rustdoc"></a>
<a id="1.69.0-Compatibility-Notes"></a>
<a id="1.69.0-Internal-Changes"></a>

Please read [the release note from some other fancy place](https://example.com).
```

And it could a link to the Git tag.

File: RELEASES.md [hunk 1]
```diff
@@ -2048,10494 +2048,7 @@ and related tools.
...
-* [Assume `build.rs` is a build script][cargo/3664]
-* [Find workspace via `workspace_root` link in containing member][cargo/3562]
-
-Misc
-----
-
-* [Documentation is rendered with mdbook instead of the obsolete, in-tree
-  `rustbook`][39633]
-* [The "Unstable Book" documents nightly-only features][ubook]
-* [Improve the style of the sid
```
Comment (weihanglo) [inline RELEASES.md:2054]: Or just have all old links as `<a>` anchors right below this.

**Comment by est31** [comment]
This fixes the issue but it breaks all links to pre 1.56 release notes, and we also know it's not a permanent solution: eventually we'll have as much content in the new file and then we'll have to split again (admittedly, that's years out). Having one file per *release* is also not good IMO because I prefer larger files over smaller ones. My proposal: make one file per edition, this size should be constant: one for edition 2015 and pre-1.0, one for edition 2018, one for edition 2020. 

Or alternatively, move the edition 2020 stuff into `RELEASES-2020.md`, and keep everything else in the old file: the edition 2020 has the fewest releases so the least amount of links are broken (at least if you assume constant amount of links for each release which might not be true due to Rust's growth).

But of course, ideally github would improve their markdown rendering engine to e.g. use a rust based one that stays below timeout limits. Has github been contacted and asked if they can find a way to address the issue *without* breaking links? Then once edition 2023 comes, we can put the notes for that one into a new file and keep everything else in the old release notes file, keeping that one at a constant size.

**Comment by Mark-Simulacrum** [comment]
I'm going to go ahead and close this PR -- thanks for putting it up. For 1.70, we linked to the rendered copy on the git tag from the blog post, and all historical releases also have associated tags + rendered markdown attached I believe.

I've also posted https://github.com/rust-lang/rust/pull/112257 which cuts the file down by ~9% without changing the rendering. It sounds like that's not enough for it to start rendering yet, so we may not merge it, but I think it's a better interim step than something like this which breaks links (as others have mentioned).

We've also reached out to GitHub (not sure if we did previously, but definitely have now) to ask about this issue and see what we can do there or learn about the particular limits we're hitting to guide a solution.

Thanks for doing this work!

**Pull request closed without merging**